frontend/src/components/drop-down-box/DropDownBox.tsx

```diff
@@ -14,17 +15,33 @@ export type DropDownBoxProps = {
   padding?: CssLength;
 };
 
-const DropDownBox: React.FC<DropDownBoxProps> = ({ children, onClose: handleClose, padding, ...positions }) => {
+const DropDownBox: React.FC<DropDownBoxProps> = ({ children, isOpen, onClose, padding, ...positions }) => {
+  const ref = useRef<HTMLDivElement>(null);
+
   useEffect(() => {
+    if (!isOpen || !ref.current) return;
+
+    const handleClose = (event: MouseEvent) => {
+      if (event.target === null) return;
+      if (event.target === ref.current) return;
+
+      const path = event.composedPath();
+      const isDropDownBoxClicked = path.findIndex(el => el === ref.current) > 0;
```

Comment:
```suggestion
      const isDropDownBoxClicked = path.some(el => el === ref.current);
```

그런데 이걸 하면 위에서 `if (event.target === ref.current) return;`가 의미가 있나요??

Reply:
반복문을 덜 돌게 하려고 처리한 건가요?

Reply:
path를 순회한것은, DropDownBox가 클릭된것인지, 그 외의 공간이 클릭된것인지 확인하기 위함입니다 :D  
그래서 DropDownBox외의 다른 공간을 클릭하면 
`!isDropDownBoxClicked && onClose();` 로직에 의해 DropDownBox가 close됩니다.

Reply:
아 0을 포함한 게 아니라 0보다 큰 경우군요. 
드롭박스 내부를 클릭하면 클로즈되지 않고 외부를 클릭했을 때만 클로즈되는 로직인 건 이해했어요!

그런데 아래 로직은
`if (event.target === ref.current) return;` 
=> `const isDropDownBoxClicked = path.some(el => el === ref.current);`이거나  `const isDropDownBoxClicked = path.findIndex(el => el === ref.current) >= 0;`이거에 포함되지 않나요?

Reply:
테스트해보니까 아래 코드도 작동 돼요!
```ts
const handleClose = (e) => {
  if (e.target === null) return;

  const path = e.composedPath();
  const isDropDownBoxClicked = path.some(el => el === ref.current);

  !isDropDownBoxClicked && onClose();
};
```

Reply:
오 맞네요, 태태가 어떤 말을 하려했는지 이해했어요.  
제안해주신 코드로 수정했습니다 :D  

[77b0b46](https://github.com/woowacourse-teams/2022-moamoa/pull/358/commits/77b0b4601df40a5ac07c54c64a377e539e731c5a)